docs/decisions/0005-architecture-and-data-modeling.rst

```diff
@@ -0,0 +1,149 @@
+
+Consequences
+************
+
+#. **New Components in the Open edX Ecosystem**: There are several new components introduced as part of this architecture:
+   - Policy Store: A centralized datastore for storing and managing authorization policies.
+   - Enforcement API: An API for enforcing authorization policies and making authorization decisions.
+   - Policy Management API: An API for creating, updating, and deleting dynamic policies in the Policy Store.
+   - Open edX Layer: The layer that abstracts access to the Policy Store and provides a unified interface for authorization.
+   - Authorization Engine: The Casbin-based engine that evaluates authorization requests based on defined policies.
+
+#. **Services Should be Migrated to Use this new Architecture**: Existing services that currently implement their own authorization logic will need to be migrated to use the new architecture. This may involve:
+   - Refactoring code to remove direct authorization checks and replace them with calls to the Enforcement API.
+   - Defining policies in the ``authz.policy`` file and/or via the Policy Management API.
+   - Ensuring that all necessary context is provided when making authorization requests.
+
+#. **The Framework Requires Client Integration**: To make authorization decisions clients must:
+   - Call the Open edX Layer's Enforcement API via network with a valid token for authentication and authorization.
+   - Install the necessary libraries (e.g., openedx-authz) as dependencies to interact with the Open edX Layer directly if they are in-process clients via the Public API (``api.py``). These libraries should be part of the services's ``INSTALLED_APPS`` -for data migration purposes- and requirements files.
+
+#. **Configuration Files Should be Deployed with Services**: For the framework to function correctly, a couple of configuration files need to be deployed alongside the services:
+   - ``model.conf``: This file contains the Casbin model configuration, defining the structure of policies and how they are evaluated. This file is essential for the Authorization Engine to understand how to process authorization requests. By default, all services share the same ``model.conf`` to ensure consistency in authorization logic across the platform.
+   - ``authz.policy``: This file contains the default policies for the service. If no policy store is configured, the service will not have any policies so no authorization decisions can be properly made.
+
+#. **Use of MySQL for Policy Storage**: The choice of MySQL as the backend for the policy store means that all policies will be stored in a relational database, which may have implications for performance and scalability. However, this also allows us to leverage existing database infrastructure and tools for managing and querying policies. If support for other databases is needed in the future, we can explore using other Casbin adapters.
```

Comment:
Not sure how much it matters, but if we're managing these things using the Django ORM we may be able to more specifically say "the LMS database", which can theoretically be Postgres too at this time.